src/target/riscv/riscv-013.c

```diff
@@ -2237,10 +2237,18 @@ static int sample_memory_bus_v1(struct target *target,
 		if (result != ERROR_OK)
 			return result;
```

Comment:
(Unrelated to this change)

It seems there is `riscv_batch_free(batch);` missing before the `return`.

Reply:
Fixed. Thanks.